src/main/java/seedu/duke/storage/Storage.java

```diff
@@ -60,17 +62,27 @@ public void writeDataOntoSaveFile(String filePath, ArrayList<String> data) throw
     public void writeTaskList(TaskList taskList) throws IOException {
         ArrayList<Task> tasks = taskList.getTaskList();
         ArrayList<String> data = StorageTasks.tasksToData(tasks);
-        writeDataOntoSaveFile(STORAGE_FILEPATH_SCHEDULE, data);
+        writeDataOntoSaveFile(StorageTasks.filePath, data); //author ngnigel99
     }
 
     //save file--->ArrayList<String> data--->ArrayList<Task> tasks
     //Can do the same for other save files by replacing the filepath constant
     public TaskList readTaskList() throws NullPointerException, IOException {
-        ArrayList<String> data = loadDataFromSaveFile(STORAGE_FILEPATH_SCHEDULE);
-        ArrayList<Task> tasks = StorageTasks.dataToTask(data);
-        TaskList tasksList = new TaskList();
-        for (int i = 0; i < tasks.size(); i++) {
-            tasksList.addTask(tasks.get(i));
+        ArrayList<Task> tasks;
+        try {
+            StorageFood.checkAndAddDirectory(StorageTasks.folderName);  //author ngnigel99
+            //ArrayList<String> data = loadDataFromSaveFile(STORAGE_FILEPATH_SCHEDULE);
+            ArrayList<String> data = loadDataFromSaveFile(StorageTasks.filePath);
+            tasks = StorageTasks.dataToTask(data);
+            TaskList tasksList = new TaskList();
+            for (int i = 0; i < tasks.size(); i++) {
+                tasksList.addTask(tasks.get(i));
+            }
+            return tasksList;
+        } catch (FileNotFoundException e) {     //author ngnigel99 - create new file
+            File f = new File(StorageTasks.filePath);
+            System.out.println("Hey, I didn't find " + StorageTasks.fileName + " in " + StorageTasks.folderName + "!");
+            System.out.println("creating new file...");
```

Comment:
@swatimahadevan here, I checked `FileNotFoundException` on creating a new file for new users, and used a variable in `Storage` rather than `Messages`. Was wondering if you could shift some of your specific operations into `StorageTasks` instead.

Reply:
for example, write and read task list because they are specific to a task list